### PR TITLE
Task02

### PR DESCRIPTION
**Pull request by decebaldecebal**

### CODE REVIEW

File: src/main/java/meeseeks/box/repository/UserRepository.java [hunk 1]
```diff
@@ -13,5 +13,6 @@
 
 @Repository
 public interface UserRepository extends CrudRepository<UserEntity, Integer> {
+    UserEntity findByUsername(final String username);
```
Comment (alexandrustoica) [inline src/main/java/meeseeks/box/repository/UserRepository.java]: When you find something always return an optional.... you will use the function ifPresent from the optional to do work on the userEntity. — see Optional<T> in Java 1.8

File: src/main/java/meeseeks/box/domain/UserEntity.java [hunk 2]
```diff
@@ -114,23 +130,39 @@ public void setUsername(final @NotNull String username) {
         this.username = username;
     }
 
-    public void setCreated(final @NotNull Calendar created) {
-        this.created = created;
-    }
```
Comment (alexandrustoica) [inline src/main/java/meeseeks/box/domain/UserEntity.java]: This should return an optional, this way you don’t have to check for null when you’re using this function.

File: src/main/java/meeseeks/box/domain/UserEntity.java [hunk 3]
```diff
@@ -114,23 +130,39 @@ public void setUsername(final @NotNull String username) {
...
         return this.name;
     }
 
     @Override
+    @JsonIgnore
     public Collection<? extends GrantedAuthority> getAuthorities() {
-        return Collections.singletonList(new SimpleGrantedAuthority("ROLE_USER"));
+    	List<GrantedAuthority> auths = new ArrayList<>();
+
+    	if (this.getRole() != null) {
```
Comment (alexandrustoica) [inline src/main/java/meeseeks/box/domain/UserEntity.java]: See... checking for null sucks. ;) Use optional to get rid of the problem.

File: src/main/java/meeseeks/box/domain/UserEntity.java [hunk 4]
```diff
@@ -114,23 +130,39 @@ public void setUsername(final @NotNull String username) {
...
+    	    auths.add(new SimpleGrantedAuthority(this.getRole().toString()));
+    	}
+
+        // All users should also have the default user role
+        if (this.getRole() != UserRole.user) {
```
Comment (alexandrustoica) [inline src/main/java/meeseeks/box/domain/UserEntity.java]: Use .equals(X)

File: src/main/java/meeseeks/box/controller/UserController.java [hunk 5]
```diff
@@ -12,25 +23,41 @@
...
+            return (UserEntity) authentication.getPrincipal();
+        }
+
         return userRepository.findByEmail("test@email.com")
                 .orElseGet(this::generateAndSaveToDatabase);
     }
 
+    @RequestMapping(value = "/current", method = RequestMethod.GET)
+    @ResponseBody
+    public UserEntity currentUserName(Authentication auth) {
```
Comment (alexandrustoica) [inline src/main/java/meeseeks/box/controller/UserController.java]: I guess this is for testing right?

File: src/main/java/meeseeks/box/controller/UserController.java [hunk 6]
```diff
@@ -12,25 +23,41 @@
...
+    public UserController(UserRepository userRepository, UserService userService) {
         this.userRepository = userRepository;
+        this.userService = userService;
     }
 
     @RequestMapping(method = RequestMethod.GET)
     public @ResponseBody UserEntity index() {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+
+        if (!(authentication instanceof AnonymousAuthenticationToken)) {
```
Comment (alexandrustoica) [inline src/main/java/meeseeks/box/controller/UserController.java]: Try ¬ to check the type of the variabile in the future. There must be an alternative way to do this.

File: src/main/java/meeseeks/box/domain/UserEntity.java (hunk 4 above)
Comment (decebaldecebal) [reply src/main/java/meeseeks/box/domain/UserEntity.java]: It's an enum, == works ok.

File: src/main/java/meeseeks/box/repository/UserRepository.java (hunk 1 above)
Comment (decebaldecebal) [reply src/main/java/meeseeks/box/repository/UserRepository.java]: Seems to complicated for our use case :)

**Pull request merged**